phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/installations/InstallationsUtils.java

```diff
@@ -0,0 +1,131 @@
+/**
+ *
+ */
+package org.phoenicis.javafx.views.mainwindow.installations;
+
+import org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationCategoryDTO;
+import org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.Function;
+
+import static org.phoenicis.configuration.localisation.Localisation.tr;
+
+public class InstallationsUtils {
+    private final static Logger LOGGER = LoggerFactory.getLogger(InstallationsUtils.class);
+
+    /**
+     * adds a new installation to an existing list of installations
+     * @param list existing list of installations
+     * @param toAdd new installation
+     * @return new list of installations containing the existing and new installations
+     */
+    public List<InstallationCategoryDTO> addInstallationToList(List<InstallationCategoryDTO> list,
+            InstallationDTO toAdd) {
+        String newInstallationCategory = toAdd.getCategory().toString();
+        final InstallationCategoryDTO newCategory = new InstallationCategoryDTO.Builder()
+                .withId(newInstallationCategory)
+                .withName(tr(newInstallationCategory))
+                .withInstallations(Collections.singletonList(toAdd))
+                .build();
+
+        final SortedMap<String, InstallationCategoryDTO> mergedCategories = new TreeMap<>(
+                createSortedMap(list, InstallationCategoryDTO::getId));
+
+        if (mergedCategories.containsKey(newCategory.getId())) {
+            mergedCategories.put(newCategory.getId(),
+                    mergeCategories(mergedCategories.get(newCategory.getId()), newCategory));
+        } else {
+            mergedCategories.put(newCategory.getId(), newCategory);
+        }
+
+        final List<InstallationCategoryDTO> categories = new ArrayList<>(mergedCategories.values());
+        categories.sort(InstallationCategoryDTO.nameComparator());
+        return categories;
+    }
+
+    /**
+     * removes as installation from an existing list of installations
+     * @param list existing list of installations
+     * @param toRemove installation which shall be removed
+     * @return new list of installations containing the existing installations without the installation which shall be removed
+     */
+    public List<InstallationCategoryDTO> removeInstallationFromList(List<InstallationCategoryDTO> list,
```

Comment:
Can you tell me how to do it with streaming API? (I'm not that familiar with it).